examples/apache-maven/custom-pipelines-hello-world/src/main/java/co/decodable/examples/cpdemo/DataStreamJob.java

```diff
@@ -19,21 +22,28 @@
 
 import co.decodable.sdk.pipeline.DecodableStreamSink;
 import co.decodable.sdk.pipeline.DecodableStreamSource;
+import co.decodable.sdk.pipeline.metadata.SinkStreams;
+import co.decodable.sdk.pipeline.metadata.SourceStreams;
 
+@SourceStreams(PURCHASE_ORDERS_STREAM)
+@SinkStreams(PURCHASE_ORDERS_PROCESSED_STREAM)
```

Comment:
That's a nice solution!

The downside is of course that you have to define the stream names at compile time .. but anything else is orders of magnitudes more complex to implement in the backend.

Reply:
Not necessarily compile time, rather assembly time. The contract is the _META-INF/decodable/stream-names.properties_ file, not the annotations. In addition, we could allow overriding at deployment or activation time, but I'd cross that bridge when there's actual demand for this raised by users.

Ultimately, I think the SDK should report stream names dynamically at runtime, but again we can improve this story iteratively.